yarn-project/pxe/src/pxe.ts

```diff
@@ -751,10 +751,26 @@ export class PXE {
         };
 
         this.log.debug(`Proving completed in ${totalTime}ms`, { timings });
-        return new TxProvingResult(privateExecutionResult, publicInputs, clientIvcProof!, {
+
+        const txProvingResult = new TxProvingResult(privateExecutionResult, publicInputs, clientIvcProof!, {
           timings,
           nodeRPCCalls: contractFunctionSimulator?.getStats().nodeRPCCalls,
         });
+
+        // TODO(benesjan): What to do with this inefficiency? Could we just return the Tx directly from here?
+        // const txHash = (await txProvingResult.toTx()).getTxHash();
```

Comment:
In a followup PR I will need to obtain a tx hash here which will most likely require some refactorings.

Reply:
I don't understand, what's the inefficiency?